saml-core/src/main/java/org/keycloak/saml/processing/api/saml/v2/sig/SAML2Signature.java

```diff
@@ -210,18 +208,20 @@ public Node getNextSiblingOfIssuer(Document doc) {
      *
      * @param document SAML document to have its ID attribute configured.
      */
-    private void configureIdAttribute(Document document) {
+    public static void configureIdAttribute(Document document) {
         // Estabilish the IDness of the ID attribute.
-        document.getDocumentElement().setIdAttribute(ID_ATTRIBUTE_NAME, true);
+        configureIdAttribute(document.getDocumentElement());
 
         NodeList nodes = document.getElementsByTagNameNS(JBossSAMLURIConstants.ASSERTION_NSURI.get(),
                 JBossSAMLConstants.ASSERTION.get());
 
         for (int i = 0; i < nodes.getLength(); i++) {
-            Node n = nodes.item(i);
-            if (n instanceof Element) {
```

Comment:
If the Document implementation is not according to spec, and the cast to Element fails, we then would be aware of and able to fix the bug in the implementation. If we do not trust the implementation to work, why bother calling the method - it might 'not work' in countless ways. The code should be safe, as there is no logical loophole (an exception will always occor). 

So unless there is a concrete example of an implemenation which triggers an exception in the code, on which the project is dependant and which cannot be fixed, keeping the old code does not make sense.